modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -4341,22 +4341,42 @@ protected void logUpdate(GridCacheOperation op, CacheObject val, GridCacheVersio
     }
 
     /**
-     * Appends a new rollback record to the write-ahead log if persistence enabled.
-     * This record keeps track the gap that corresponds to the update counter of out-of-order update in the ATOMIC cache.
+     * Appends a new out-of-order update record to the write-ahead log if persistence enabled.
+     * This record keeps track the out-of-order updates in the ATOMIC cache.
      *
-     * @param gapStart Update counter corresponds to out-of-order update.
+     * @param op Update operation.
+     * @param val Write value.
+     * @param writeVer Write version.
+     * @param expireTime Expire time.
+     * @param updCntr Update counter.
      */
-    protected void logOutOfOrderUpdate(Long gapStart) throws IgniteCheckedException {
+    protected void logOutOfOrderUpdate(
```

Comment:
We should check whether WAL is enabled when we call this method.

Reply:
This check is already added below.